charts/aspnetcore/templates/deployment-hpa.yaml

```diff
@@ -0,0 +1,24 @@
+{{- if (((.Values.deployment).autoscaling).enabled) }}
+apiVersion: autoscaling/v2
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ .Release.Name }}-hpa
+  annotations:
+    {{- with .Values.commonAnnotations }}
+      {{- toYaml . | nindent 4 }}
+    {{- end }}
+spec:
+  metrics:
+    - resource:
+        name: cpu
+        target:
+          averageUtilization: {{ .Values.deployment.autoscaling.targetCPUUtilizationPercentage }}
+          type: Utilization
+      type: Resource
+  minReplicas: {{ max (add ((.Values.podDisruptionBudget).minAvailable) 1) (.Values.deployment.autoscaling.minReplicas) }}
```

Comment:
I'm wondering if it's better to have it like this or just assign the value directly and let users be mindful of the value they set with regard to HPA and PDB. @starcraft66 any thoughts on this?

Reply:
I went in with the midset of this is Workleap base image. Hpa are currently not in templates but those templates are used in production, a bad value mean a potential sirt. which is avoidable.